src/components/NewClusterModal.js

```diff
@@ -219,6 +219,22 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
     })
   }
 
+  sendUpdateMetrics(newRuntime, newPersistentDisk) {
+    const machineType = newRuntime.cloudService === cloudServices.GCE ? newRuntime.machineType : newRuntime.masterMachineType
+    const { cpu, memory } = findMachineType(machineType)
+
+    Ajax().Metrics.captureEvent(Events.cloudEnvironmentUpdate, {
+      ...extractWorkspaceDetails(this.makeWorkspaceObj()),
+      ...newRuntime,
+      cpu,
+      memory,
+      persistentDiskSize: newPersistentDisk?.size,
+      runtimeCostPerHour: Utils.formatUSD(runtimeConfigCost(this.getPendingRuntimeConfig())),
+      runtimePausedCostPerHour: Utils.formatUSD(ongoingCost(this.getPendingRuntimeConfig())),
+      persistentDiskCostPerMonth: (newPersistentDisk && Utils.formatUSD(persistentDiskCostMonthly(newPersistentDisk)))
```

Comment:
There's a `getPendingDisk` method that we should use to feed the cost calculation now. Passing `newPersistentDisk` technically works for the moment, but it's the wrong type. The cost functions expect leo-shaped objects, whereas the environment config stuff is an internal representation.